### ISSUE TITLE
[Feature] Add Intel QSV H265 

### ISSUE THREAD

**Issue by Serph91P** [MrBrax/LiveStreamDVR#304]
You already have the Reencode integrated and tried to use nvidia. Would there be a possiblity to add Intel QSV H265?
https://trac.ffmpeg.org/wiki/Encode/H.265

**Comment by MrBrax** [comment]
> ffmpeg needs to be built with the --enable-gpl --enable-libx265 configuration flags

this is always the larger problem, i'm guessing it needs a lot of libraries to build

**Comment by Serph91P** [comment]
> > ffmpeg needs to be built with the --enable-gpl --enable-libx265 configuration flags
> 
> this is always the larger problem, i'm guessing it needs a lot of libraries to build, i guess this one was just for h265 though
> 
> i tried to combine the nvidia container and mine but it didn't work

Dunno if I am wrong but this seems pretty straight forward for me.
https://ffmpeg.org/ffmpeg-all.html#Intel-QuickSync-Video

**Comment by MrBrax** [comment]
so you need to build your own ffmpeg binary and then attach your cpu or whatever via docker commands, that's a bit above my understanding, and i don't even use linux/intel

**Comment by Serph91P** [comment]
> so you need to build your own ffmpeg binary and then attach your cpu or whatever via docker commands, that's a bit above my understanding, and i don't even use linux/intel

yeah how to get the gpu to the docker is quite easy. The only thing is ffmpeg to be build with those commands as far as I understand. Got there from the other page I linked in frist post. Where it said: You can also [download a static build](https://ffmpeg.org/download.html), all of which bundle libx265. So its already in there with those static builds.

**Comment by MrBrax** [comment]
so it should be in the installed one in the docker container already? i have no way of testing this, so not really any practical way of doing it

**Comment by Serph91P** [comment]
> so it should be in the installed one in the docker container already? i have no way of testing this, so not really any practical way of doing it

will look into it and report back.

**Comment by MrBrax** [comment]
also seems like the command line arguments for it are double that of regular encodes, bloating the selection up even more

**Comment by Serph91P** [comment]
> also seems like the command line arguments for it are double that of regular encodes, bloating the selection up even more

Would it be bad if I encoded the files with another application?

**Comment by MrBrax** [comment]
bad? uh not really, but replacing the segments would still keep the old mediainfo and such

**Comment by Serph91P** [comment]
> bad? uh not really, but replacing the segments would still keep the old mediainfo and such

ok will try and see what is happening.
